dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/engine/task/statemachine/TaskSubmittedStateAction.java

```diff
@@ -109,7 +110,12 @@ public void onDispatchEvent(final IWorkflowExecutionRunnable workflowExecutionRu
                     taskInstance.getDelayTime(),
                     remainTimeMills);
         }
-        taskExecutionRunnable.initializeTaskExecutionContext();
+        try {
+            taskExecutionRunnable.initializeTaskExecutionContext();
+        } catch (Exception ex) {
+            log.error("Current taskInstance: {} initializeTaskExecutionContext fail", taskInstance.getName(), ex);
+            throw new TaskFatalException(taskInstance.getName() + " initializeTaskExecutionContext fail");
+        }
```

Comment:
```suggestion

            taskExecutionRunnable.initializeTaskExecutionContext();
```

It's better to throw `TaskFatalException` in `TaskExecutionContextFactory#createTaskExecutionContext` method. Not all exception should be transformed to `TaskFatalException`.

Reply:
> taskExecutionRunnable.initializeTaskExecutionContext();

ok，catch exception and publish TaskFatalLifecycleEvent in doFireSingleWorkflowEventBus